frontend/src/components/ui/data-grid/controllers/rows.ts

```diff
@@ -15,19 +22,22 @@ import { cached } from "@/utils/weakCache";
  * that are slotted into `<btrix-data-grid>`, it may be necessary to
  * implement this controller on the container component.
  */
-export class DataGridRowsController implements ReactiveController {
+export class DataGridRowsController<Item = GridItem>
```

Comment:
I thiiiink there's a way we could enforce a `RowKey` generic type on `DataGrid` so that it's checked that calls to `setItems` etc have the correct type with a row id present, but for now just adding `extends GridItem` is an easier little improvement to type check this a little better.
```suggestion
export class DataGridRowsController<Item extends GridItem = GridItem>
```

Reply:
Nice, makes sense!